error-message-guidelines.md

```diff
@@ -0,0 +1,383 @@
+---
+layout: global
+title: Error message guidelines
+type: "page singular"
+navigation:
+  weight: 5
+  show: true
+---
+
+## Error Message Guidelines
+
+This guide is a reference for composing standardized and actionable
+error messages in Apache Spark.
+
+### Include What, Why, and How
+
+Exceptions thrown from Spark should answer the Five W's and How:
```

Comment:
Although we know `where` means the stack trace, please add `**Where**` in the following list. Currently, we say `the Five W's` and show only four Ws in the list. So, this mismatch unnecessarily confuse readers.

Reply:
Whoops, good catch. Thanks!